proto/hypertrace/agent/config/v1/config.proto

```diff
@@ -36,6 +36,9 @@ message AgentConfig {
 
     // telemetry
     Telemetry telemetry = 8;
+
+    // blocking config
+    BlockingConfig blocking_config = 9;
```

Comment:
Why is this in the hypertrace agent-config?

Reply:
javaagent needs it in ht. We have blocking logic in instrumented code in ht. Evaluation logic whether to block is in traceable.

Reply:
I agree this looks weird too.

Reply:
The ht javaagent will only read the ht config, and we have the instrumentation logic in ht javaagent. That is where we do blocking. A filter only evaluates whether to block or not, but the actual instrumentation where we block the request is in ht. So we need this config at that place in ht during blocking. 
We can change the filter interface to also return this blocking code and msg along with blocking evaluation from filters. 
But since we anyway have the blocking logic in ht javaagent, I feel adding this to ht would serve same purpose with less complication of going through filters.
Also, I guess this would be the case in goagent as well?

Reply:
No, we didn't need this change for goagent. Can you sync with @prodion23?

Reply:
@shashank11p In short, goagent gets around this by updating the filter api to return a FilterResult instead of a simple bool to indicate block. 

see here in traceable/goagent: 
https://github.com/traceableai/goagent-src/blob/2bc7f832f01e9149b826212578cbc0b042b479e3/filter/traceable/filter.go#L363-L366

& then in ht/goagent it is just reads the statuscode from filter result. 

I think the above is essentially the same as what you were suggesting:
> We can change the filter interface to also return this blocking code and msg along with blocking evaluation from filters.

(I'm a fan of keeping the configs independent to blocking for traceable vs tracing features for ht

Reply:
We can do that, and even in javaagent we kept the blocking evaluations and configs as much in traceable, but if ht is the one doing blocking and anyway getting the blocking info, then why not directly give it to ht instead of making it come from filters?
Overall, if we look at it, It seems like we are giving ht a way to configure the blocking status code and msg, but not directly, and we would have made it complicated thinking about traceable(to keep blocking config in traceable), and not thinking about ht directly(while it has that capability)

Reply:
I had kept the config exactly same as traceable config, so we would not need 2 blocking configs. Only 1, which would be read by both.
But,
> The ht agent would never use that since no filters are present. Only when a traceable agent is used, which I think indicates this config more so belongs to traceable agent-config.

Thanks. This cleared up everything. Since blocking won't happen unless there is a filter that tells it to block, that filter can also tell the blocking code and msg. 
We do not need this PR.